spec/src/main/asciidoc/configprofile.asciidoc

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright (c) 2020 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// Contributors:
+// Emily Jiang
+
+
+[[configprofile]]
+== Config Profile
+
+Config Profile indicates the project phase, such as dev, testing, live, etc. 
+=== Specify Config Profile
+
+The config profile can be specified via a system property `mp.config.profile`. It can be set when starting your microservice, e.g.
```

Comment:
We should also allow builders to set the profile for the Config they are building. Using a system property is nice on a global level but it doesn't make sense if an application uses a custom config or even more custom configs. 

A new method like this in `ConfigBuilder` would allow it:

```
ConfigBuilder withProfile(String profileName)
```

Reply:
It ought to be possible to set the config profile via a configuration property IMO.

Reply:
Since when reading config sources, we need to figure out the config profile value. Using system properties is much simplier. What will happen if set different value for `mp.config.profile` in different config sources, which makes the situation very complicated.

Reply:
Highest config source wins, as usual.  We do support self-configuration in other areas, so I don't think this is so unusual.